packages/relay/src/lib/eth.ts

```diff
@@ -825,4 +825,96 @@ export class EthImpl implements Eth {
         return null;
       });
   }
+
+  async getLogs(blockHash: string|null, fromBlock: string|null, toBlock: string|null, address: string|null, topics: any[]|null): Promise<Log[]> {
+    const params: any = {};
+    if (blockHash) {
+      const block = await this.mirrorNodeClient.getBlock(blockHash);
+      if (block) {
+        params.timestamp = [
+          `gte:${block.timestamp.from}`,
+          `lte:${block.timestamp.to}`
+        ];
+      }
+    }
+    else if (fromBlock && toBlock) {
+      const blocksResult = await this.mirrorNodeClient.getBlocks([
+          `gte:${fromBlock}`,
+          `lte:${toBlock}`
+      ]);
+
+      const blocks = blocksResult?.blocks;
+      if (blocks?.length) {
+        const firstBlock = blocks[0];
+        const lastBlock = blocks[blocks.length - 1];
+        params.timestamp = [
+          `gte:${firstBlock.timestamp.from}`,
+          `lte:${lastBlock.timestamp.to}`
+        ];
+      }
+    }
+
+    if (topics) {
+      for (let i = 0; i < topics.length; i++) {
+        params[`topic${i}`] = topics[i];
+      }
+    }
+
+    let result;
+    if (address) {
+      result = await this.mirrorNodeClient.getContractResultsLogsByAddress(address, params);
+    }
+    else {
+      result = await this.mirrorNodeClient.getContractResultsLogs(params);
+    }
+
+    if (!result || !result.logs) {
+      return [];
+    }
+    const logs = result.logs;
+
+    // Find all unique contractId and timestamp pairs and for each one make mirror node request
+    const promises: Promise<any>[] = [];
+    const uniquePairs: string[] = [];
```

Comment:
Not sure how large logs count could be so better to use a dictionary lookup instead
```suggestion
    const uniquePairs = {};
```
Then later on check for `uniquePairs[pair] === undefined` for presence. 
On absence you can just set it equal to true or some valid but small default value